src/sentry/incidents/endpoints/serializers/workflow_engine_detector.py

```diff
@@ -1,34 +1,350 @@
+
+    def add_created_by(
+        self, result: DefaultDict[Detector, dict[str, Any]], detectors: Sequence[Detector]
+    ) -> None:
+        user_by_user_id: MutableMapping[int, RpcUser] = {
+            user.id: user
+            for user in user_service.get_many_by_id(
+                ids=[
+                    detector.created_by_id
+                    for detector in detectors
+                    if detector.created_by_id is not None
+                ]
+            )
+        }
+        for detector in detectors:
+            # this is based on who created or updated it during dual write
+            rpc_user = None
+            if detector.created_by_id:
+                rpc_user = user_by_user_id.get(detector.created_by_id)
+            if not rpc_user:
+                result[detector]["created_by"] = {}
+            else:
+                created_by = dict(
+                    id=rpc_user.id, name=rpc_user.get_display_name(), email=rpc_user.email
+                )
+                result[detector]["created_by"] = created_by
+
+    def add_owner(
+        self, result: DefaultDict[Detector, dict[str, Any]], detectors: Sequence[Detector]
+    ) -> None:
+        for detector in detectors:
+            if detector.owner_user_id or detector.owner_team_id:
+                actor = detector.owner
+                if actor:
+                    result[detector]["owner"] = actor.identifier
+
+    def add_latest_incident(
+        self,
+        result: DefaultDict[Detector, dict[str, Any]],
+        user: User | RpcUser | AnonymousUser,
+        detectors: dict[int, Detector],
+        detector_to_action_ids: defaultdict[Detector, list[int]],
+    ) -> None:
+        all_action_ids = []
+        for action_ids in detector_to_action_ids.values():
+            all_action_ids.extend(action_ids)
+
+        action_group_statuses = ActionGroupStatus.objects.filter(action_id__in=all_action_ids)
+
+        detector_to_group_ids = defaultdict(list)
+        for action_group_status in action_group_statuses:
+            for detector, action_ids in detector_to_action_ids.items():
+                if action_group_status.action_id in action_ids:
+                    detector_to_group_ids[detector].append(action_group_status.group_id)
+
+        open_periods = None
+        group_ids = [action_group_status.group_id for action_group_status in action_group_statuses]
+        if group_ids:
+            open_periods = GroupOpenPeriod.objects.filter(
+                group__in=[group_id for group_id in group_ids]
+            )
+
+        for detector in detectors.values():
+            # TODO: this serializer is half baked
+            if open_periods:
+                open_periods.filter(Q(group__in=detector_to_group_ids[detector])).order_by(
+                    "-date_started"
+                )
+
+                serialized_group_open_period = serialize(
+                    open_periods[0], user, WorkflowEngineIncidentSerializer()
+                )
```

Comment:
i don't believe the `.filter` is an in place operation here -- i think we'd need to do something like:
```suggestion
                latest_open_periods = open_periods.filter(Q(group__in=detector_to_group_ids[detector])).order_by(
                    "-date_started"
                )

                serialized_group_open_period = serialize(
                    latest_open_periods.first(), user, WorkflowEngineIncidentSerializer()
                )
```